src/lib/builders/tree/types.ts

```diff
@@ -0,0 +1,19 @@
+import type { createTreeViewBuilder  } from "./create";
+
+// TODO: add multiselect option?
+export type CreateTreeViewArgs = {
+	forceVisible?: boolean;
+	// multiselect?: boolean;
+};
+
+export type TreeParts =
+	| 'label'
+	| 'item'
+	| 'group';
+
+export type ItemDescription = {
+    hasChildren: boolean;
+    childrenIdxs: number[];
+};
+
+export type CreateTreeViewReturn = ReturnType<typeof createTreeViewBuilder>;
```

Comment:
There are more types needed here, take a look at the other builders to see what they export

Reply:
I also added the elements, states, and helper types now. Are there other types that also need to be exported?